### DISCUSSION TITLE
Win32 API: retrieving scroll bar information

### DISCUSSION THREAD

**Discussion by g055s00** [microsoft/terminal#16351] [Q&A] [answered]
How can I retrieve scroll bar information in Windows Terminal? GetScrollBarInfo is returning only STATE_SYSTEM_INVISIBLE for rgstate[0]

**Comment by DHowett** [comment]
This will probably not work because Windows Terminal uses XAML for its internal UI elements. Even if it didn't, though, finding the correct set of scrollbars for the active pane inside the active tab is going to be a bit of a chore.

What's your use case? Is there a way we could make that easier?

**Reply by g055s00** [reply]
I'm developing a TUI using PowerShell, and I need to control the behavior of scroll bars in both PowerShell and Windows Terminal. The objective is to hide the scroll bars if they're visible and restore them when the script exits. In PowerShell, adjusting the buffer size achieves this, but in Windows Terminal, it seems to have only a persistent scroll bar setting, which is not affected by the buffer size.

**Reply by zadjii-msft** [reply, marked as answer]
Ah. You probably want to use the [alternate screen buffer](https://learn.microsoft.com/en-us/windows/console/console-virtual-terminal-sequences#alternate-screen-buffer). This is the thing that apps like `vim`, `emacs`, `tmux` et al use to temporarily create a "full screen" application in the terminal, then dismiss the buffer when they're done. 

The alt buffer will dismiss the scrollbars for you 😉
